CONTRIBUTING.md

```diff
@@ -0,0 +1,267 @@
+tree shaking can still help optimize application start time amongst other things.
+Technically speaking, Lux is not built until a project that uses Lux is built—so why
+build in development at all? Well, some parts of Lux core are required to be built and
+executable in the targeted Node.js version(s).
+
+To build the required modules for local development, execute the following commands:
+
+```bash
+# Clone this repository from GitHub.
+git clone https://github.com/postlight/lux.git
+
+# Navigate into the root of this repository.
+cd lux
+
+# Install local dependencies.
+npm install
+
+# Clean any build artifacts remaining from prior builds. This step is optional
+# and only makes sense if a build has already occurred.
+npm run clean
+
+# Run the build tools.
+npm run build
+
+# Symlink the lux-framework to the global node_modules directory. This ensures
+# that the "lux" command will use the files we just built instead of another
+# install.
+npm link
+```
+
+For general debugging and sanity checking code changes to Lux core, you can use
+the [test app](./test/test-app). To run the test app for the first time,
+execute the following commands:
+
+```bash
+# Navigate to the test/test-app directory from the root of this repository.
+cd test/test-app
+
+# Create the database schema.
+lux db:create
+
+# Run any pending database migrations.
+lux db:migrate
+
+# Seed the database with fixtures.
+lux db:seed
+
+# Run the application server.
+lux serve
+```
+
+### Testing
+
+First make sure you have built lux locally and it is symlinked to your global
+`node_modules` directory. Once you have completed the build steps, execute the
+following command:
+
+```bash
+# Run the test suite from the root of this repository.
+NODE_ENV="test" npm test
+```
+
+*NOTE:*
+
+Subsequent executions of the test suite can run without going through the build
+steps again.
+
+### Code Style
+
+#### Flow
+
+The Lux codebase is statically typed with [Flow](https://flowtype.org). This
+helps prevent many common bugs including ones caused by [unexpected type coercions](https://www.destroyallsoftware.com/talks/wat).
+
+If you have never written JavaScript with [Flow](https://flowtype.org) or
+[TypeScript](https://www.typescriptlang.org/) before, you may have a bit of a
+learning curve. If you get stuck on something, feel free to ask a question in the
+[*postlight/lux*](https://gitter.im/postlight/lux) and/or [*facebook/flow*](https://gitter.im/facebook/flow)
+Gitter room. We recommend installing an editor plugin with autocompletion and
+realtime error checking. Below you will find a few example plugins for many
+common editors.
+
+*   [Nuclide (Atom)](https://nuclide.io/)
+*   [FlowIDE (Sublime Text)](https://github.com/tptee/FlowIDE)
+*   [flow-for-vscode (Visual Studio Code)](https://github.com/flowtype/flow-for-vscode)
+*   [flow-for-emacs (Emacs)](https://github.com/flowtype/flow-for-emacs)
+*   [vim-flow (Vim)](https://github.com/flowtype/vim-flow)
+
```

Comment:
Maybe add WebStorm too? I feel that with that one included you've covered the most popular text-editors/IDE's. Unfortunately the link is version specific (though official flow support has only been added in that (the most recent) version).

Link: https://www.jetbrains.com/help/webstorm/2016.3/using-the-flow-type-checker.html